injectproxy/silences.go

```diff
@@ -0,0 +1,158 @@
+package injectproxy
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"path"
+	"strconv"
+	"strings"
+
+	runtimeclient "github.com/go-openapi/runtime/client"
+	"github.com/go-openapi/strfmt"
+	"github.com/prometheus/alertmanager/api/v2/client"
+	"github.com/prometheus/alertmanager/api/v2/client/silence"
+	"github.com/prometheus/alertmanager/api/v2/models"
+	"github.com/prometheus/alertmanager/pkg/labels"
+)
+
+func (r *routes) silences(w http.ResponseWriter, req *http.Request) {
+	switch req.Method {
+	case "GET":
+		r.listSilences(w, req)
+	case "POST":
+		r.postSilence(w, req)
+	}
```

Comment:
can we quickly document here the expected behavior in case `req.Method != "GET" && req.Method != "POST"` ?

Reply:
Agreed. Though it shouldn't happen because of the new HTTP routing logic (https://github.com/openshift/prom-label-proxy/pull/20/files#diff-70a97a8db748a07af11564c1ac089aabR36-R37), it's more clear to have it called out explicitly.

Reply:
I guess the edge case that we cover here now, is when the set of methods declared in `enforceMethods` does not match the set of handled methods here. Thanks for handling it :+1: